src/annotator/util/test/extract-annotation-query-test.js

```diff
@@ -2,93 +2,117 @@
-    },
-    {
-      describe: "when there's a non-alphanumeric annotation ID",
-      it: 'returns null',
-      url: 'http://localhost:3000#annotations:not%20alphanumeric',
-      returns: null,
-    },
-
-    // #annotations:query:<QUERY> fragments, in their various forms.
-    {
-      describe: "when there's a #annotations:query:<QUERY> fragment",
-      it: 'returns an object containing the query',
-      url: 'http://localhost:3000#annotations:query:user:fred',
-      returns: {query: 'user:fred'},
-    },
-    {
-      describe: "when there's a #annotations:q:<QUERY> fragment",
-      it: 'returns an object containing the query',
-      url: 'http://localhost:3000#annotations:q:user:fred',
-      returns: {query: 'user:fred'},
-    },
-    {
-      describe: "when there's a #annotations:QuerY:<QUERY> fragment",
-      it: 'returns an object containing the query',
-      url: 'http://localhost:3000#annotations:QuerY:user:fred',
-      returns: {query: 'user:fred'},
-    },
-    {
-      describe: 'when the query contains both a username and a tag',
-      it: 'returns an object containing the query',
-      url: 'http://localhost:3000#annotations:q:user:fred%20tag:foo',
-      returns: {query: 'user:fred tag:foo'},
-    },
-    {
-      describe: 'when the query contains URI escape sequences',
-      it: 'decodes the escape sequences',
-      url: 'http://localhost:3000#annotations:query:foo%20bar',
-      returns: {query: 'foo bar'},
-    },
-
-    // Unrecognised or missing #fragments.
-    {
-      describe: "when there's an unrecognised URL fragment",
-      it: 'returns null',
-      url: 'http://localhost:3000#unknown',
-      returns: null,
-    },
-    {
-      describe: "when there's no URL fragment",
-      it: 'returns null',
-      url: 'http://localhost:3000',
-      returns: null,
-    },
-  ].forEach(function(test) {
-    describe(test.describe, function() {
-      it(test.it, function() {
-        assert.deepEqual(extractAnnotationQuery(test.url), test.returns);
+describe('annotation.util.extractAnnotationQuery', function() {
+  describe('#annotations', function() {
+    [
+      {
+        describe: "when there's a valid #annotations:<ID> fragment",
+        it: 'returns an object containing the annotation ID',
+        url: 'http://localhost:3000#annotations:alphanum3ric_-only',
+        returns: 'alphanum3ric_-only',
+      },
+      {
+        describe: "when there's a non-alphanumeric annotation ID",
+        it: 'returns null',
+        url: 'http://localhost:3000#annotations:not%20alphanumeric',
+        returns: null,
+      },
+      {
+        describe: "when there's an unrecognised URL fragment",
+        it: 'returns null',
+        url: 'http://localhost:3000#unknown',
+        returns: null,
+      },
+      {
+        describe: "when there's no URL fragment",
+        it: 'returns null',
+        url: 'http://localhost:3000',
+        returns: null,
+      },
```

Comment:
The last two test cases here are duplicated in the tests for the other function below. The actual test code test code is almost exactly duplicated as well. I don't think this is a big deal. The duplication could be removed by some cleverness in the test code, but I don't think that would be worth the cost.

If we grow any more config settings that are read from the URL fragment (or maybe from anywhere in the URL) in the future, then it might be worth adding something like a generic `settingFromUrl(name, url)` or `settingFromUrlFragment(name, url)` function that can be unit tested just once and reused.